docs: Document Auto-configure repository AI feature (OD-420) (#2737)
codacy/docs: docs/codacy-ai/codacy-ai.md

%}

<!-- TODO: verify plan tier. The GA release checklist (OD-577) still had "is it only for paid?" as an open question at doc-writing time. -->

_This feature uses an AI coding agent, and is strictly opt-in: it only runs when an organization admin starts it for a specific repository._

Auto-configure repository sets up Codacy analysis for a repository without the manual steps in [Configuring your repository](../getting-started/configuring-your-repository.md). An AI agent reads the repository, then applies settings such as which tools and code patterns to enable, which paths to exclude, and how to configure each language — the same settings you could otherwise set by hand or through a [Codacy configuration file](../repositories-configure/codacy-configuration-file.md).

**What happens during a run**

1. Codacy clones the repository into an isolated environment, using credentials scoped and time-limited to that one run.
2. An AI agent reviews the repository's structure, languages, and frameworks.
3. The agent applies a Codacy configuration based on what it finds.
4. Codacy re-analyzes the repository using the new configuration.

**How to turn it on**

> 1. Go to the repository's dashboard in Codacy.

Reviewer: this should be: repository's coding patterns page
right?

Reviewer 2: +1 @LipeGheno

2. Select <!-- TODO: verify exact button/menu label and where it lives in the dashboard --> **Auto-configure repository**. Only organization admins see this option.
3. Start a run, then follow its status (queued, running, successful, or failed) from the dashboard.
4. Once a run finishes, select <!-- TODO: verify exact label --> **Review** to see a summary of what changed, including the languages and files the agent analyzed.

> **Notes**

Reviewer 2: I would add to notes smth like: 
 Autoconfig takes several minutes. You can leave this page — the results appear here when the run finishes.

> - Only organization admins can start a run.

Reviewer: we should also say that we only allow 1 run per day

Reviewer 2: yes 
1 run per day per repo

- Codacy limits how many runs happen at the same time, both for the organization and across Codacy, so runs stay fast for everyone. If you start a run while one is already queued or running for the same repository, Codacy queues it instead of starting it immediately.
- Each run has a fixed amount of time and AI usage it can use, so a run always finishes or fails instead of continuing indefinitely.
- You can start a new run at any time; it replaces the configuration from the previous run.

> - The agent only has access to the repository's code and to the credentials needed for that one run. Codacy discards both when the run ends.

Reviewer 3: This could probably be something like below:
```suggestion
- The agent only has access to the repository's code for the duration of that one run. It is immediately discarded when the run ends.
```

- To enrich the review, the git diff of the Pull Request as well as some related files' contents can be sent as context. No data is stored on our side, or used to train any models.
- Prompts are neither stored nor visible to anyone.
- As an extra precaution, before any code snippet is sent to the AI model, Codacy automatically redacts secrets (API keys, tokens, credentials, and other high-entropy strings) from the code context.

### Auto-configure repository {: id="auto-configure-repository"}

{%
    include-markdown "../assets/includes/paid.md"
    start="<!--paid-feature-start-->"
    end="<!--paid-feature-end-->"
%}

<!-- TODO: verify plan tier. The GA release checklist (OD-577) still had "is it only for paid?" as an open question at doc-writing time. -->

_This feature uses an AI coding agent, and is strictly opt-in: it only runs when an organization admin starts it for a specific repository._

> Auto-configure repository sets up Codacy analysis for a repository without the manual steps in [Configuring your repository](../getting-started/configuring-your-repository.md). An AI agent reads the repository, then applies settings such as which tools and code patterns to enable, which paths to exclude, and how to configure each language — the same settings you could otherwise set by hand or through a [Codacy configuration file](../repositories-configure/codacy-configuration-file.md).

Reviewer 4: i think /repositories-configure/configuring-code-patterns/ is a better page than ../getting-started/configuring-your-repository.md

Reviewer 4: i also think we could add a section on ../getting-started/configuring-your-repository.md with the juicy details. this codacy AI page is more of a legal handoff we can provide to stakeholders

Reviewer 4: > which paths to exclude

auto config doesnt exclude anything. it recommends improvements on your configurations.

Reviewer 4: > applies settings such as which tools and code patterns to enable, which paths to exclude, and how to configure each language

i also find this relatively generic. how to configure each language? what?

Suggestion:
> An AI agent understands the requirements of the repository, and then curates settings such as which tools and code patterns to enable, and provides you recommendations on coding standards and paths to exclude to get a more accurate analysis.

**What happens during a run**

> 1. Codacy clones the repository into an isolated environment, using credentials scoped and time-limited to that one run.

Reviewer 4: > using credentials scoped and time-limited to that one run
is this relevant for the end user? isnt isolated environment enough?

2. An AI agent reviews the repository's structure, languages, and frameworks.
3. The agent applies a Codacy configuration based on what it finds.
4. Codacy re-analyzes the repository using the new configuration.

**How to turn it on**

1. Go to your repository's [Code patterns](../repositories-configure/configuring-code-patterns.md) page in Codacy.

    ![The Code patterns page, with the repository's issue count and the Auto-configure button](images/auto-configure-code-patterns.png)

2. Select **Auto-configure**. Only organization admins see this option.

    ![The Auto-configure button on the Code patterns page](images/auto-configure-button.png)

3. Start a run, then follow its status (queued, running, successful, or failed) from the dashboard.
4. Once a run finishes, select <!-- TODO: verify exact label --> **Review** to see a summary of what changed, including the languages and files the agent analyzed.

**Auto-configure repository notes**

> - Only organization admins can start a run.

Reviewer 4: Needs to be reviewed to "Anyone with repository permissions to edit patterns can run auto-configuration". we can even potentially send users to a place where roles are decribed.

> - You can start at most one run per repository per day.

Reviewer 4: Each repository can be auto-configured at most once a day

> - A run takes several minutes. You can leave this page — the results appear here when the run finishes.

Reviewer 4: - A run will take several minutes. You can leave the page — the results appear on the Patterns page once the run finishes.

- Codacy limits how many runs happen at the same time, both for the organization and across Codacy, so runs stay fast for everyone. If you start a run while one is already queued or running for the same repository, Codacy queues it instead of starting it immediately.
- Each run has a fixed amount of time and AI usage it can use, so a run always finishes or fails instead of continuing indefinitely.
- You can start a new run at any time; it replaces the configuration from the previous run.

> - The agent only has access to the repository's code for the duration of that one run. It is immediately discarded when the run ends.

Reviewer 4: Last things i would also add:
- During the autoconfiguration your repo's metrics and issues may fluctuate.
- Once auto-configuration is over you will be able to review all changes made and recommendations provided.

im also not sure if in this section there are other legal notes we need to add, for example, which model (?) or that none of your code is being used to train anything etc etc etc
